### PR TITLE
add VRTX supply queries

### PR DESCRIPTION
**Pull request by yruej301**

### CODE REVIEW

File: packages/indexer-client/src/IndexerBaseClient.ts [hunk 1]
```diff
@@ -814,6 +816,20 @@ export class IndexerBaseClient {
...
 
+  /**
+   * Retrieves VRTX total / circulating supply
+   *
+   * @param params
+   */
+  async getVrtxTokenInfo(
+    params: GetIndexerVrtxTokenInfoParams,
+  ): Promise<GetIndexerVrtxTokenInfoResponse> {
+    const baseUrl = this.opts.url.replace('/v1', '');
```
Comment (frankfka) [inline packages/indexer-client/src/IndexerBaseClient.ts:827]: Should we have some sort of `this.v2Url` for when we add more queries in the future?

We should address types as well but can do that later.

Would we be able to just replace /v1 with /v2?

File: packages/indexer-client/src/types/clientTypes.ts [hunk 2]
```diff
@@ -676,3 +677,9 @@ export interface IndexerLeaderboardContest {
 export interface GetIndexerLeaderboardContestsResponse {
   contests: IndexerLeaderboardContest[];
 }
+
+export interface GetIndexerVrtxTokenInfoParams {
+  tokenInfoType: IndexerVrtxTokenInfoType;
+}
+
+export type GetIndexerVrtxTokenInfoResponse = number;
```
Comment (frankfka) [inline packages/indexer-client/src/types/clientTypes.ts:685]: Is it a number or a string from backend?

Comment (frankfka) [reply packages/indexer-client/src/types/clientTypes.ts:685]: Is this in X18? ie is it 10^9+18 or just 10^9 being returned?

File: packages/indexer-client/src/IndexerBaseClient.ts (hunk 1 above)
Comment (yruej301) [reply packages/indexer-client/src/IndexerBaseClient.ts:827]: yea, will do!

File: packages/indexer-client/src/types/clientTypes.ts (hunk 2 above)
Comment (yruej301) [reply packages/indexer-client/src/types/clientTypes.ts:685]: it's just a number and it's 10^9

File: packages/indexer-client/src/IndexerBaseClient.ts (hunk 1 above)
Comment (yruej301) [reply packages/indexer-client/src/IndexerBaseClient.ts:827]: addressed!

File: packages/indexer-client/src/IndexerBaseClient.ts [hunk 3]
```diff
@@ -109,6 +112,7 @@ export class IndexerBaseClient {
   readonly axiosInstance: AxiosInstance;
 
   constructor(opts: IndexerClientOpts) {
+    opts.v2Url = opts.v2Url ? opts.v2Url : opts.url.replace('v1', 'v2');
```
Comment (frankfka) [inline packages/indexer-client/src/IndexerBaseClient.ts:115]: Why would we need the override? Would it be safe to just derive this internally for now?

Not sure we should override opts, I feel like it should be immutable as much as possible.

Could we store this in a readonly variable similar to axiosInstance?

File: packages/indexer-client/src/types/clientTypes.ts (hunk 2 above)
Comment (frankfka) [reply packages/indexer-client/src/types/clientTypes.ts:685]: Can we leave a JSDoc comment indicating that this does not include decimals?

File: packages/indexer-client/src/IndexerBaseClient.ts (hunk 3 above)
Comment (yruej301) [reply packages/indexer-client/src/IndexerBaseClient.ts]: check now

**Comment by yruej301** [comment]
> Can we leave a JSDoc comment indicating that this does not include decimals?

@frankfka do you mean, that is not x18? it does include decimals, see sample responses

```
circulating supply: 244529235.8614397
total supply: 997999999.9999979
```

**Comment by frankfka** [comment]
Decimals as in token decimals, ig im not sure what exactly to call it, but in FE when something includes decimals, it has additional 0s corresponding to the decimals of the token

Ex 1 USDC with decimals is 10^6

**Comment by yruej301** [comment]
gotcha

**Pull request merged**